From 4f62e0e4efe6e1ecd5ea657d76b23657b3c09cb9 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Tue, 3 Dec 2019 01:03:49 -0800
Subject: [PATCH 0063/1851] firmware: xilinx: Add qspi firmware interface

commit 859b94639ef1312aad46c54c529b895f241f4ce4 from
https://github.com/Xilinx/linux-xlnx.git

Add support for QSPI ioctl functions and enums.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Rajan Vaja <rajan.vaja@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/firmware/xilinx/zynqmp.c     |  1 +
 include/linux/firmware/xlnx-zynqmp.h | 15 ++++++++++++++-
 2 files changed, 15 insertions(+), 1 deletion(-)

diff --git a/drivers/firmware/xilinx/zynqmp.c b/drivers/firmware/xilinx/zynqmp.c
index 58bde69611cd..f662da17799f 100644
--- a/drivers/firmware/xilinx/zynqmp.c
+++ b/drivers/firmware/xilinx/zynqmp.c
@@ -518,6 +518,7 @@ static inline int zynqmp_is_valid_ioctl(u32 ioctl_id)
 	case IOCTL_GET_PLL_FRAC_MODE:
 	case IOCTL_SET_PLL_FRAC_DATA:
 	case IOCTL_GET_PLL_FRAC_DATA:
+	case IOCTL_SET_TAPDELAY_BYPASS:
 	case IOCTL_SET_SGMII_MODE:
 	case IOCTL_WRITE_GGS:
 	case IOCTL_READ_GGS:
diff --git a/include/linux/firmware/xlnx-zynqmp.h b/include/linux/firmware/xlnx-zynqmp.h
index c7e879661300..f6f298c254ff 100644
--- a/include/linux/firmware/xlnx-zynqmp.h
+++ b/include/linux/firmware/xlnx-zynqmp.h
@@ -144,7 +144,8 @@ enum pm_ret_status {
 
 enum pm_ioctl_id {
 	IOCTL_GET_RPU_OPER_MODE,
-	IOCTL_SET_SGMII_MODE = 5,
+	IOCTL_SET_TAPDELAY_BYPASS = 4,
+	IOCTL_SET_SGMII_MODE,
 	IOCTL_SET_PLL_FRAC_MODE = 8,
 	IOCTL_GET_PLL_FRAC_MODE,
 	IOCTL_SET_PLL_FRAC_DATA,
@@ -388,6 +389,18 @@ enum zynqmp_pm_shutdown_subtype {
 	ZYNQMP_PM_SHUTDOWN_SUBTYPE_SYSTEM,
 };
 
+enum tap_delay_signal_type {
+	PM_TAPDELAY_NAND_DQS_IN,
+	PM_TAPDELAY_NAND_DQS_OUT,
+	PM_TAPDELAY_QSPI,
+	PM_TAPDELAY_MAX,
+};
+
+enum tap_delay_bypass_ctrl {
+	PM_TAPDELAY_BYPASS_DISABLE,
+	PM_TAPDELAY_BYPASS_ENABLE,
+};
+
 enum sgmii_mode {
 	PM_SGMII_DISABLE,
 	PM_SGMII_ENABLE,
-- 
2.31.1

